### PR TITLE
Added Copy Button in Qr Code

### LINKED ISSUE #11: Add 'copy link' to share options
there's a "share" button when viewing a session
currently it opens a modal with QR code (see below)

desired behaviour: add a "copy link" with a nice button section below the qr code, which will put the link in clipboard

### PR DESCRIPTION
**Pull request by Ronald-patrick**
Added Copy Button in QR code !  #11  ( closes #11 )

### CODE REVIEW

**Comment by tarnas14** [comment]
hey, thank you for your contribution! it works exactly as I imagined :)

some feedback:

1. please remove `package-lock.json` from the PR, there's already a `yarn.lock`, because `yarn` is now the standard package manager for create-react-app (which was used to start the client repo)

more general advice:
- when you're working on an issue, you can reference it in commit message or PR description, like this: #11 <- this will link the PR to the issue. (additionally, if you include text `closes #11`, merging the PR will automatically close the issue)
- it's generally a good practice in OSS to describe in the PR description what has been done. For UI-related tasks it is also nice to provide a screenshot or a recording of the result

**Review by tarnas14 — CHANGES_REQUESTED** [review]
please remove `package-lock.json` from the PR, there's already a `yarn.lock`, because `yarn` is now the standard package manager for create-react-app (which was used to start the client repo)

**Comment by Ronald-patrick** [comment]
Thanks for the Guidance, I understood how the github works now.

**Comment by Ronald-patrick** [comment]
Deleted package-lock.json.

File: client/src/SessionView/ShareButton.js [hunk 1]
```diff
@@ -21,6 +23,13 @@ function ShareButton({
   const classes = useStyles()
   const [showQr, setShowQr] = useState(false)
 
+	const copyLink = (e)=>{
+		navigator.clipboard.writeText(`${window.location.origin}/${id}`);
+		const copyButton = e.target;
+		copyButton.innerText = "Copied!";
+		console.log(e.target);
```
Comment (tarnas14) [inline client/src/SessionView/ShareButton.js:30]:
```suggestion
```

please remove console.log :) sorry I missed it in the previous pass

**Review by tarnas14 — CHANGES_REQUESTED** [review]
whoops, I missed the console.log yesterday, but this is a perfect opportunity for using a 'suggestion' in comment - you can suggest changes in PRs and github allows committing them without leaving the browser :D
please commit the suggestion and I will merge the PR

once again thank you for your contribution!

**Comment by Ronald-patrick** [comment]
Wow that was awesome !

**Comment by Ronald-patrick** [comment]
Also i wanted to ask that it's giving me the same error as before if i run the react app using my terminal, But when i run in docker it is running fine, but the problem is i have to restart the whole react app in docker to see the changes i made in code as the changes are not reflected in page directly it takes too much time.

**Review by tarnas14 — APPROVED** [review]

**Comment by tarnas14** [comment]
// I approved the PR, I'll merge it when we finish the docker discussion

what OS are you using?
the docker-compose and react docker are made to watch file changes made on the system, so no restart should be required
is the hot-reload not working at all, or does it take time?

**Comment by Ronald-patrick** [comment]
I am using windows

**Comment by Ronald-patrick** [comment]
Hot restart is happening but it takes too much time for a single change

**Comment by Ronald-patrick** [comment]
Thank you for Approving my PR !

**Comment by tarnas14** [comment]
could you prepare [an issue](https://github.com/tarnas14/ti4-companion/issues/new) - explain your setup and what problems you are experiencing?

I'm not saying I'll fix it, because I'm using linux, but if you have time and motivation you could try and solve it and prepare another PR :D

**Comment by Ronald-patrick** [comment]
Actually it is a local issue, I there is an actual need I'll add an issue . Also Can you merge the pull request ?

**Comment by tarnas14** [comment]
merging right now, I wanted to finish the discussion about your docker problems, thank you for your contribution <3

**Pull request merged**